bmc-log-collector/log-pointer.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"path"
+)
+
+type LastPointer struct {
+	Serial       string
+	LastReadTime int64
+	LastReadId   int
+}
+
+func (c *logCollector) readLastPointer(serial string, ptrDir string) (LastPointer, error) {
+	var lptr LastPointer
+
+	c.mutex.Lock()
```

Comment:
Should not lock all files in one mutex. This may cause performance problems.

It should be locked file by file. Or lock may not be necessary.

Reply:
fix